packages/flutter/lib/src/material/stepper.dart

```diff
@@ -0,0 +1,679 @@
+    return new Container(
+      margin: const EdgeInsets.symmetric(vertical: 8.0),
+      width: _kStepSize,
+      height: _kStepSize,
+      child: new Center(
+        child: new SizedBox(
+          width: _kStepSize,
+          height: _kTriangleHeight, // Height of 24dp-long-sided equilateral triangle.
+          child: new CustomPaint(
+            painter: new _TrianglePainter(
+              color: _isDark() ? _kErrorDark : _kErrorLight
+            ),
+            child: new Align(
+              alignment: const FractionalOffset(0.5, 0.9), // 0.9 looks better than the geometrical 0.66.
+              child: _buildCircleChild(index, oldState && config.steps[index].state != StepState.error)
+            )
+          )
+        )
+      )
+    );
+  }
+
+  Widget _buildIcon(int index) {
+    if (config.steps[index].state != _oldStates[index]) {
+      return new AnimatedCrossFade(
+        firstChild: _buildCircle(index, true),
+        secondChild: _buildTriangle(index, true),
+        firstCurve: new Interval(0.0, 0.6, curve: Curves.fastOutSlowIn),
+        secondCurve: new Interval(0.4, 1.0, curve: Curves.fastOutSlowIn),
+        sizeCurve: Curves.fastOutSlowIn,
+        crossFadeState: config.steps[index].state == StepState.error ? CrossFadeState.showSecond : CrossFadeState.showFirst,
+        duration: kThemeAnimationDuration,
+      );
+    } else {
+      if (config.steps[index].state != StepState.error)
+        return _buildCircle(index, false);
+      else
+        return _buildTriangle(index, false);
+    }
+  }
+
+  Widget _buildVerticalControls() {
+    Color cancelColor;
+
+    switch (Theme.of(context).brightness) {
+      case Brightness.light:
+        cancelColor = Colors.black54;
+        break;
+      case Brightness.dark:
+        cancelColor = Colors.white70;
+        break;
+    }
+
+    assert(cancelColor != null);
+
+    final ThemeData theme = Theme.of(context);
+
+    return new Container(
+      margin: const EdgeInsets.only(top: 16.0),
+      child: new ConstrainedBox(
+        constraints: const BoxConstraints.tightFor(height: 48.0),
+        child: new Row(
+          children: <Widget>[
+            new FlatButton(
+              onPressed: config.onStepContinue,
+              color: theme.brightness == Brightness.light ? theme.primaryColor : theme.backgroundColor,
+              textColor: Colors.white,
+              textTheme: ButtonTextTheme.normal,
+              child: new Text('CONTINUE')
+            ),
+            new Container(
+              margin: const EdgeInsets.only(left: 8.0),
+              child: new FlatButton(
+                onPressed: config.onStepCancel,
+                textColor: cancelColor,
+                textTheme: ButtonTextTheme.normal,
+                child: new Text('CANCEL')
+              )
+            )
+          ]
+        )
+      )
+    );
+  }
+
+  TextStyle _titleStyle(int index) {
+    ThemeData themeData = Theme.of(context);
```

Comment:
This variable should be defined consistently throughout this file. Although we're not consistent throughout flutter, I think the most sensible variation is:

`final ThemeData themeData = Theme.of(context);`